### PR TITLE
Getting link acceptance test 

### PR DESCRIPTION
**Pull request by sitaramireddytanuboddi**
happy path for link

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -3,12 +3,25 @@ import sbt.*
 object Dependencies {
 
   val test: Seq[ModuleID] = Seq(
-    "com.typesafe"         % "config"                  % "1.4.3"  % Test,
-    "com.typesafe.play"   %% "play-ahc-ws-standalone"  % "2.2.5"  % Test,
-    "com.typesafe.play"   %% "play-ws-standalone-json" % "2.2.5"  % Test,
-    "com.vladsch.flexmark" % "flexmark-all"            % "0.64.8" % Test,
-    "org.scalatest"       %% "scalatest"               % "3.2.18" % Test,
-    "org.slf4j"            % "slf4j-simple"            % "2.0.9"  % Test
+    // "com.typesafe"            % "config"                  % "1.4.3"  % Test,
```
Comment (szaroug-hmrc-2024) [inline project/Dependencies.scala:6]: delete all comment out line please

File: src/test/resources/application.conf [hunk 2]
```diff
@@ -15,38 +15,62 @@
 environment: local
 
 local {
+     api-name: "http://localhost:9260"
+     api-platform-test-user: "http://localhost:9260"
+     api-platform-agents-auth: ""
+     api-platform-agents-auth-test: ""
+     auth-login-stub_uri: "/government-gateway/session/login"
+     redirect-url: "http://localhost:9949/auth-login-stub/session"
+     emcs-api-uri: "movements"
```
Comment (szaroug-hmrc-2024) [inline src/test/resources/application.conf:24]: please remove anything that is not part our api config

File: src/test/resources/application.conf [hunk 3]
```diff
@@ -15,38 +15,62 @@
...
+    }
+    ims {
+      port = 9653
+      productionRoute = ""
     }
+    emcs{
+        port = 10250
+        productionRoute = ""
+    }
+    tfc{
```
Comment (szaroug-hmrc-2024) [inline src/test/resources/application.conf:42]: refer to this as tfcp

File: src/test/resources/test-push-poll-setup-draft.sh
Comment (szaroug-hmrc-2024) [inline src/test/resources/test-push-poll-setup-draft.sh:1]: we may need this sh file later down the line
for now remove it please

File: src/test/scala/uk/gov/hmrc/test/api/client/RestAssured.scala [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  def getRequestSpec: RequestSpecification = {
+    val requestSpec = given()
+      .config(config().headerConfig(headerConfig().overwriteHeadersWithName("Authorization", "Content-Type")))
+      .contentType(ContentType.XML)
+      .baseUri(url)
+    initiateProxy(requestSpec)
+    requestSpec
+  }
+
+//  def clearQueryParam(requestSpecification: RequestSpecification): Unit = {
```
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/client/RestAssured.scala:44]: remove commented out code please

File: src/test/scala/uk/gov/hmrc/test/api/helpers/IndividualsMatchingHelper.scala
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/helpers/IndividualsMatchingHelper.scala:1]: do we need this

File: src/test/scala/uk/gov/hmrc/test/api/models/IndividualsLinks.scala
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/models/IndividualsLinks.scala:1]: do we need this?

File: src/test/scala/uk/gov/hmrc/test/api/models/User.scala
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/models/User.scala:1]: remove this please

File: src/test/scala/uk/gov/hmrc/test/api/service/ExciseMovementControlSystemService.scala
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/service/ExciseMovementControlSystemService.scala:1]: is this needed??
its refering to ExciseMovementControlSystem

File: src/test/scala/uk/gov/hmrc/test/api/service/IndividualsMatchingService.scala
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/service/IndividualsMatchingService.scala:1]: is that needed?

File: src/test/scala/uk/gov/hmrc/test/api/specs/BaseSpec.scala [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import org.scalatest.concurrent.Eventually
+import org.scalatest.featurespec.AnyFeatureSpec
+import org.scalatest.matchers.should.Matchers
+import uk.gov.hmrc.test.api.helpers.{AuthHelper, ExciseMovementControlSystemHelper, IndividualsMatchingHelper, TestDataHelper}
+
+trait BaseSpec extends AnyFeatureSpec with GivenWhenThen with BeforeAndAfterAll with Matchers with Eventually {
+  val authHelper                        = new AuthHelper
+  val testDataHelper                    = new TestDataHelper
+  val individualsMatchingHelper         = new IndividualsMatchingHelper
+  val exciseMovementControlSystemHelper = new ExciseMovementControlSystemHelper
```
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/specs/BaseSpec.scala:29]: is this correct?

File: src/test/scala/uk/gov/hmrc/test/api/specs/CommonSpec.scala
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/specs/CommonSpec.scala:1]: review the code on this file.
as its not clear what is the intention

File: src/test/scala/uk/gov/hmrc/test/api/specs/CommonSpec.scala [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+      .andReturn()
+  }
+  def tfcLink(token: String): Response                                                    =
+    requestSpecification
+      .header("Authorization", token)
+      .header("Content-Type", "application/json")
+      .header("Accept", "application/vnd.hmrc.1.0+json")
+      .when()
+      .body(
+        "{\"correlationId\":\"AAAA-BBBB-CCCC-DDDD\", \"epp_unique_customer_id\":\"EPP-ID\", \"epp_reg_reference\":\"EPP-Req-Ref\", \"outbound_child_payment_ref\":\"Out-Bound-Child-Ref\", \"child_date_of_birth\":\"01-02-2023\"}"
```
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/specs/CommonSpec.scala:244]: please, use GUID as in this format
5c5ef9c2-72e8-4d4f-901e-9fec3db8c64b

File: src/test/scala/uk/gov/hmrc/test/api/utils/DateUtils.scala
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/utils/DateUtils.scala:1]: remove this file please if we dont use/need it yet

File: src/test/scala/uk/gov/hmrc/test/api/utils/GenerateClientIdAndBoxId.scala
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/utils/GenerateClientIdAndBoxId.scala:1]: remove as it not related to this PR

File: src/test/scala/uk/gov/hmrc/test/api/utils/RemoveJsonElement.scala
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/utils/RemoveJsonElement.scala:1]: remove this please

File: src/test/scala/uk/gov/hmrc/test/api/utils/TextUtils.scala
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/utils/TextUtils.scala:1]: remove please

File: src/test/scala/uk/gov/hmrc/test/api/utils/Zap.scala
Comment (szaroug-hmrc-2024) [inline src/test/scala/uk/gov/hmrc/test/api/utils/Zap.scala:1]: remove please

File: src/test/resources/application.conf (hunk 2 above)
Comment (sitaramireddytanuboddi) [reply src/test/resources/application.conf]: done

File: src/test/resources/application.conf [hunk 7]
```diff
@@ -15,38 +15,62 @@
...
+    auth {
+      port = 8585
+#       port = 9949
+      productionRoute = ""
+    }
+    ims {
+      port = 9653
+      productionRoute = ""
     }
+    emcs{
```
[the remark this answers is not in the document]
Comment (sitaramireddytanuboddi) [reply src/test/resources/application.conf]: Done

File: src/test/resources/application.conf (hunk 3 above)
Comment (sitaramireddytanuboddi) [reply src/test/resources/application.conf]: done

File: src/test/resources/test-push-poll-setup-draft.sh
Comment (sitaramireddytanuboddi) [reply src/test/resources/test-push-poll-setup-draft.sh:1]: done

File: src/test/scala/uk/gov/hmrc/test/api/client/RestAssured.scala (hunk 4 above)
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/client/RestAssured.scala]: done

File: src/test/scala/uk/gov/hmrc/test/api/helpers/ExciseMovementControlSystemHelper.scala
[the remark this answers is not in the document]
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/helpers/ExciseMovementControlSystemHelper.scala:1]: done

File: src/test/scala/uk/gov/hmrc/test/api/helpers/IndividualsMatchingHelper.scala
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/helpers/IndividualsMatchingHelper.scala:1]: removed

File: src/test/scala/uk/gov/hmrc/test/api/helpers/TestDataHelper.scala
[the remark this answers is not in the document]
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/helpers/TestDataHelper.scala:1]: removed

File: src/test/scala/uk/gov/hmrc/test/api/models/IndividualsLinks.scala
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/models/IndividualsLinks.scala:1]: removed

File: src/test/scala/uk/gov/hmrc/test/api/models/User.scala
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/models/User.scala:1]: done

File: src/test/scala/uk/gov/hmrc/test/api/service/AuthService.scala [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class AuthService(filename: Any) extends HttpClient {
+
+  val host: String        = TestConfiguration.url("auth")
+  val redirectUrl: String = TestConfiguration.getConfigValue("redirect-url")
+
+  def authPayload(identifier: String): String =
```
[the remark this answers is not in the document]
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/service/AuthService.scala]: removed unused method

File: src/test/scala/uk/gov/hmrc/test/api/service/ExciseMovementControlSystemService.scala
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/service/ExciseMovementControlSystemService.scala:1]: removed

File: src/test/scala/uk/gov/hmrc/test/api/specs/BaseSpec.scala (hunk 5 above)
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/specs/BaseSpec.scala]: updated

File: src/test/scala/uk/gov/hmrc/test/api/specs/CommonSpec.scala
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/specs/CommonSpec.scala:1]: updated

File: src/test/scala/uk/gov/hmrc/test/api/specs/TfcAPIs.scala [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.test.api.specs
+
+import uk.gov.hmrc.test.api.client.HttpClient
+
+class TfcAPIs extends BaseSpec with CommonSpec with HttpClient {
+
+  Feature("Excise TFC APIs") {
```
[the remark this answers is not in the document]
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/specs/TfcAPIs.scala]: done

File: src/test/scala/uk/gov/hmrc/test/api/utils/DateUtils.scala
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/utils/DateUtils.scala:1]: removed

File: src/test/scala/uk/gov/hmrc/test/api/utils/GenerateClientIdAndBoxId.scala
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/utils/GenerateClientIdAndBoxId.scala:1]: removed

File: src/test/scala/uk/gov/hmrc/test/api/utils/JsonUtils.scala
[the remark this answers is not in the document]
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/utils/JsonUtils.scala:1]: removed

File: src/test/scala/uk/gov/hmrc/test/api/utils/LRNGenerator.scala
[the remark this answers is not in the document]
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/utils/LRNGenerator.scala:1]: removed

File: src/test/scala/uk/gov/hmrc/test/api/utils/RandomUtils.scala
[the remark this answers is not in the document]
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/utils/RandomUtils.scala:1]: done

File: src/test/scala/uk/gov/hmrc/test/api/utils/RemoveJsonElement.scala
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/utils/RemoveJsonElement.scala:1]: done

File: src/test/scala/uk/gov/hmrc/test/api/utils/TextUtils.scala
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/utils/TextUtils.scala:1]: done

File: src/test/scala/uk/gov/hmrc/test/api/utils/XmlUtils.scala
[the remark this answers is not in the document]
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/utils/XmlUtils.scala:1]: done

File: src/test/scala/uk/gov/hmrc/test/api/utils/Zap.scala
Comment (sitaramireddytanuboddi) [reply src/test/scala/uk/gov/hmrc/test/api/utils/Zap.scala:1]: done

**Review by szaroug-hmrc-2024 — APPROVED** [review]
approved to get it merged, however, the outstanding comment we could consider for the next PR.

**Review by szaroug-hmrc-2024 — DISMISSED** [review]

**Review by szaroug-hmrc-2024 — APPROVED** [review]
thanks Sitarami

**Pull request closed without merging**